### PR TITLE
=str 21049 gaps in FlowOps section of stages-overview documentation

### PR DESCRIPTION
**Pull request by agolubev**
Ref #21049

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

File: akka-docs/rst/java/stream/stages-overview.rst [hunk 1]
```diff
@@ -570,6 +570,17 @@ it returns false the element is discarded.
 
 **completes** when upstream completes
 
+filteNot
```
Comment (johanandren) [inline akka-docs/rst/java/stream/stages-overview.rst]: filte*r*Nor

File: akka-docs/rst/java/stream/stages-overview.rst [hunk 2]
```diff
@@ -630,6 +641,17 @@ complete the current value is emitted downstream.
 
 **completes** when upstream completes
 
+reduce
+^^^^^^
+Start with first element and then apply the current and next value to the given function, when upstream
+complete the current value is emitted downstream. Similar to ``fold``.
+
+**emits** when upstream completes
```
Comment (johanandren) [inline akka-docs/rst/java/stream/stages-overview.rst]: This isn't right, its more like `the result of the applied function when upstream emits an element`, no?

File: akka-docs/rst/java/stream/stages-overview.rst [hunk 3]
```diff
@@ -713,6 +735,57 @@ a function has to be provided to calculate the individual cost of each element.
...
+
+**emits** when upstream emits and the number of emitted elements has not reached max
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes and the number of emitted elements has not reached max
+
+log
+^^^
+Log elements flowing through the stream as well as completion and erroring.
```
Comment (johanandren) [inline akka-docs/rst/java/stream/stages-overview.rst]: maybe mention that the log level can be set with the attributes `Attributes.loglevels` (scala) and `Attributes.createLogLevels()` (java) ?

Comment (johanandren) [inline akka-docs/rst/java/stream/stages-overview.rst]: And what the default level is.

File: akka-docs/rst/java/stream/stages-overview.rst [hunk 4]
```diff
@@ -713,6 +735,57 @@ a function has to be provided to calculate the individual cost of each element.
...
+
+**emits** when the mapping function returns an element
```
Comment (johanandren) [inline akka-docs/rst/java/stream/stages-overview.rst]: this is copied from somewhere else, "emits when upstream emits"

File: akka-docs/rst/java/stream/stages-overview.rst (hunk 2 above)
Comment (agolubev) [inline akka-docs/rst/java/stream/stages-overview.rst]: it emits aggregated state when upstream just completed. `stream.impl.fusing.Reduce` has only one push and it's pushing aggregate state.

Or your comment was not about emits section?

Comment (johanandren) [inline akka-docs/rst/java/stream/stages-overview.rst]: No, you are correct. I mixed it up with Scan.

**Comment by johanandren** [comment]
LGTM

**Comment by drewhk** [comment]
LGTM

**Pull request merged**